### PR TITLE
Phpunit adjustments

### PR DESCRIPTION
**Pull request by johangant**
Misc tweaks to make things more obvious and less prone to breakage or clash with other tools we use.

### CODE REVIEW

File: config/phpunit.lando.xml [hunk 1]
```diff
@@ -23,13 +23,13 @@
     <!-- Do not limit the amount of memory tests take to run. -->
     <ini name="memory_limit" value="-1"/>
     <!-- Example SIMPLETEST_BASE_URL value: http://localhost -->
-    <env name="SIMPLETEST_BASE_URL" value=""/>
+    <env name="SIMPLETEST_BASE_URL" value="http://nidirect-d8.lndo.site"/>
```
Comment (omahm) [inline config/phpunit.lando.xml:26]: I'd look at dumping in the lando site url env variable as we have different lando configs/names. 
We should also add in the generic database parameters that the d8 recipe provides.

File: scripts/appserver_build.sh [hunk 2]
```diff
@@ -46,7 +46,7 @@ chmod -w /app/drupal8/web/sites/default
 sed -i -e "s|name=\"SIMPLETEST_BASE_URL\" value=\"\"|name=\"SIMPLETEST_BASE_URL\" value=\"http:\/\/${LANDO_APP_NAME}.${LANDO_DOMAIN}\"|g" /app/config/phpunit.lando.xml
 if [ -f "/app/config/phpunit.lando.xml" ]; then
   echo "Copying PHPUnit config to Drupal webroot"
-  ln -sf /app/config/phpunit.lando.xml /app/drupal8/web/core/phpunit.xml
+  cp /app/config/phpunit.lando.xml /app/drupal8/web/core/phpunit.xml
```
Comment (omahm) [inline scripts/appserver_build.sh:49]: Does linking not work? Just thinking we will run into issues if people update the config/ version

Comment (johangant) [reply scripts/appserver_build.sh:49]: It does, but I've found copying the file into the container at runtime or when you need it to change keeps it consistent with other work I've done lately. Looking at it after a sleep or two I sense there's some variation in how things are generated/passed in whether that's a literal value in a file + the entire file copied in (like this example), a symlink or `sed` to rewrite certain parts of a file then copy it in.

My gut feeling is drawing me to literal + copy in, with sed + copy in where we need a dynamic value (eg: a Lando hostname). We can't achieve the latter with a symlink unless we expect people to either keep the exact same Lando hostnames or other config, or require them to exercise discipline to not commit any host-specific changes to the repo (never works :))

**Comment by johangant** [comment]
Agree with the broader points here. Given that I'm tidying up a bunch of stuff that supersedes this I'll merge and continue to extend with a separate PR

**Pull request merged**